src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -1700,14 +1767,20 @@ private static ParameterSymbol GetCorrespondingParameter(int argumentOrdinal, Me
             }
 
             Debug.Assert((object)parameter != null || !expanded);
-            if (expanded && (parameter.Ordinal < method.ParameterCount - 1 || !parameter.Type.IsSZArray()))
+            if (expanded && (parameter.Ordinal < n - 1 || !parameter.Type.IsSZArray()))
             {
                 expanded = false;
             }
 
             return parameter;
         }
 
+        private static TypeSymbolWithAnnotations GetParameterType(ParameterSymbol parameter, bool expanded)
+        {
+            var type = parameter.Type;
+            return (expanded && parameter.IsParams) ? ((ArrayTypeSymbol)type.TypeSymbol).ElementType : type;
```

Comment:
>IsParams [](start = 42, length = 8)

Checking IsParams is not sufficient, the parameter also must be the last parameter and its type must be IsSZArray

Reply:
`GetCorrespondingParameter` was already checking those conditions and setting `expanded = false` otherwise. I've merged these two methods so that's clear, and added tests.

---
In reply to: [181204752](https://github.com/dotnet/roslyn/pull/26107#discussion_r181204752) [](ancestors = 181204752)